src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -282,7 +282,8 @@ private static void ValidateManagedIdentityResult(ManagedIdentityResponse respon
                 HandleInvalidExternalValueError(nameof(response.AccessToken));
             }
 
-            long expiresIn = DateTimeHelpers.GetDurationFromNowInSeconds(response.ExpiresOn);
+            long expiresIn = DateTimeHelpers.GetDurationFromManagedIdentityTimestamp(response.ExpiresOn);

```

Comment:
nit: there is no need to parse the expires in again. I'd just copy this "validate" private method logic into its caller.